### ISSUE TITLE
Building within Docker container

### ISSUE THREAD

**Issue by kreuzerkrieg** [facebook/fbzmq#6]
Hi,
Am I supposed to be able to build the fbzmq in docker container running vanilla Ubuntu 16.04?

Sincerely,
Ernest

**Comment by plapukhov** [comment]
Should be fine if you have all of the required dependencies (which is plentiful, including boost and folly). Any specific issues you are encountering?

**Comment by kreuzerkrieg** [comment]
Ok, regarding 16.04 container `folly/DefaultKeepAliveExecutor.h` is indeed not in /usr/local/include/folly (bad `make install` in `folly` script?) , so I've added the build/deps/folly to the include path, the `wangle` was built. Next stop, fails to build `fbthrift`, `CMake` cant find `follyConfig.cmake`. To be continued.

**Comment by plapukhov** [comment]
Ernest, it will take me some time to repro the build issue, mainly because it's full-blown work week here. I will do my best to follow up asap. I'm sorry about the pain, building those project with loads of complex dependencies is never easy :(

**Comment by kreuzerkrieg** [comment]
No rush Petr, I got it working locally, both in container and on my dev machine, so it is ok for now.

**Comment by saifhhasan** [comment]
@kreuzerkrieg Since you got working, we would encourage patches from you if you did any to address build issue with `build_fbzmq.sh`

**Comment by kreuzerkrieg** [comment]
@saifhhasan not sure the patch is good idea. And thats why:
1. I assume you are running some kind of CI and I guess it is working. It is suspicious I have to make any changes if your CI is working properly.
2. I'm not sure I did things right, the fact that I got it working does not ensure things are working properly.

In any case, I can try to create patch, but I guess it is a good idea someone takes look to the changes I've made by inspecting the above link to the gist.
